src/overlays/actors/ovl_En_Fg/z_en_fg.c

```diff
@@ -21,40 +47,418 @@ const ActorInit En_Fg_InitVars = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+static EnFgAnimation sAnimations[] = {
+    { &D_06001534, 1.0f, 0, -1, 0, 0 },
+    { &D_06001534, 1.0f, 0, -1, 0, -4 },
+    { &D_060011C0, 1.0f, 0, -1, 0, -4 },
+    { &D_060007BC, 1.0f, 0, -1, 2, -4 },
+};
+
+s32 func_80A2D280(SkelAnime* skelAnime, s16 animIndex) {
+    s16 frameCount;
+    s32 ret;
+
+    ret = false;
+    if (animIndex >= 0 && animIndex < 4) {
+        ret = true;
+        frameCount = sAnimations[animIndex].frameCount;
+        if (frameCount < 0) {
+            frameCount = SkelAnime_GetFrameCount(&sAnimations[animIndex].animationSeg->common);
+        }
+        SkelAnime_ChangeAnim(skelAnime, sAnimations[animIndex].animationSeg, sAnimations[animIndex].playbackSpeed,
+                                sAnimations[animIndex].frame, frameCount, sAnimations[animIndex].mode,
+                                sAnimations[animIndex].transitionRate);
+    }
+    return ret;
+}
+
+void func_80A2D348(EnFg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health != 0) {
+        this->collider.dim.pos.x = this->actor.world.pos.x;
+        this->collider.dim.pos.y = this->actor.world.pos.y;
+        this->collider.dim.pos.z = this->actor.world.pos.z;
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+    }
+}
+
+void func_80A2D3D4(EnFg* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+u8 EnFg_UpdateHealth(EnFg* this) {
+    if (this->actor.colChkInfo.damage >= this->actor.colChkInfo.health) {
+        this->actor.colChkInfo.health = 0;
+    } else {
+        this->actor.colChkInfo.health -= this->actor.colChkInfo.damage;
+    }
+    return this->actor.colChkInfo.health;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D348.asm")
+s32 func_80A2D42C(EnFg* this) {
+    s32 ret = 0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D3D4.asm")
+    if (this->collider.base.acFlags & 2) {
+        switch (this->actor.colChkInfo.damageEffect) {
+            case 1:
+                ret = FG_DMGEFFECT_DEKUSTICK;
+                break;
+            case 15:
+                ret = FG_DMGEFFECT_HOOKSHOT;
+                break;
+            case 14:
+                ret = FG_DMGEFFECT_ARROW;
+                break;
+            case 3:
+                ret = FG_DMGEFFECT_ICEARROW;
+                break;
+            default:
+                ret = FG_DMGEFFECT_EXPLOSION;
+                break;
+        }
+        this->collider.base.acFlags &= ~2;
+        EnFg_UpdateHealth(this);
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D400.asm")
+void func_80A2D4B8(EnFg* this, GlobalContext* globalCtx) {
```

Comment:
I'm thinking this is `EnFg_Idle`

Reply:
At least in this case I would prefer without the `ing`, since its not the idle function as the frog is `jumping`, more the idle function that causes the frog to `jump`

Reply:
Alternatively could do something like `EnFg_IdleOnGround` and `EnFg_IdleInAir`.

Reply:
Fair enough about the `ing`. I'm not sure they describe `OnGround` vs `InAir` because the difference is one sets y-velocity to +10, so it's as the frog rises, while the other sets y-velocity to 0 so it's as the frog falls

Reply:
Actually on second look at the frogs and their jumping cycle, I do think this function is just `EnFg_Idle` and the other function is the trigger for a jump, so just `EnFg_Jump`

Reply:
Yeah but when it changes the velocity, it switches to the other function. So velocity gets set to +10, the frog jumps and is in the air, and the actionfunc switches. The other one sets velocity to 0, the frog lands and is on the ground, and the actionfunc switches back.